### DISCUSSION TITLE
Use custom Certificate Manager instead of TLS sidecar in Entity Operator

### DISCUSSION THREAD

**Discussion by prateekkohli21** [strimzi/strimzi-kafka-operator#7480] [Q&A]
Hi All,

Is is possible to integrate a custom certificate manager and disable the sideCarTLS container(so that it does not get deployed) while using entity operator?

I would want to give my custom certificates and manage them using a Certificate Manager for communication between topic operator & kafka/zookeeper, but I did not see any option to disable the sidecarTLS container & do certificate management using a custom certificate Manager.

Regards,
Prateek Kohli

**Comment by scholzj** [comment]
The TLS sidecar is nothing like _Certificate Manager_. So I'm not sure in which way it would it replace it. The TLS sidecar just encrypts the connection when conencting to ZooKeeper because historically, ZooKeeper had. abad support for TLS.

**Reply by prateekkohli21** [reply]
Thanks for your response!!

So how would it be possible to specify custom certificates for cluster internal communication (for eg. between Topic operator & zookeeper.) instead of the default ones?

**Reply by scholzj** [reply]
Are you using a Strimzi managed Kafka cluster? Or standalone Topic Operator? For Strimzi managed Kafka cluster, you can provide your own custom CA. Check the docs for more details.

**Reply by prateekkohli21** [reply]
I am using Strimzi managed Kafka Cluster, will refer the documentation and discuss further in case of any concerns. 
Thanks a lot!!

**Reply by prateekkohli21** [reply]
Just wanted to know if I use standalone topic Operator, do I have an option of connection to zookeeper on TLS?

I can see STRIMZI_TLS_ENABLED option, but I guess this is just for mTLS communication between topic-operator & zookeeper.

Is there any similar option to make mTLS communication between topic-operator & zookeeper?

**Reply by scholzj** [reply]
If you use Strimzi managed Kafka cluster, then you should enable the Topic Operator through the Kafka CR and not use the standalone Topic Operator.

**Reply by prateekkohli21** [reply]
Ok, but if I want to use standalone topic operator, how can I manage TLS with external zookeeper?

**Reply by scholzj** [reply]
You cannot. If you use Strimzi based Kafka cluster, you should simply use the Topic Operator through Kafka CR.
